test/unit_tests/test_kb_ad_vina_utils.py

```diff
@@ -1,18 +1,125 @@
 import logging
+import os
 import subprocess
 
 import pytest
 
 from kb_ad_vina.utils import ADVinaApp
 
-@pytest.fixture
-def process():
-    return subprocess.Popen(
-        ["echo", "hello world"],
+
+def approximatelyEqual(float1, float2, epsilon):
+    return abs(float1 - float2) < epsilon
+
+
+def clean():
+    with subprocess.Popen(
+        ["rm *.pdbqt"],
+        shell=True,
         stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE
+        stderr=subprocess.PIPE,
+    ) as proc:
+        proc.communicate()
+
+
+EPSILON = 0.5
+MODULE_HOME = (
+    "/kb/module" if os.environ.get("KBASE_CONTAINER") == "yes" else "."
+)
+os.chdir(os.path.join(MODULE_HOME, "test/data"))
+affinity = {
+    "Structure2D_CID_49846579": -8.1,
+    "Structure2D_CID_139024764": -7.9,
+}
+clean()
+
+
+@pytest.fixture
+def ligands():
+    return sorted(affinity.keys(), reverse=True)
+
+
+@pytest.fixture
+def receptor():
+    return "6wzu"
+
+
+def test_01_pdb_to_pdbqt(receptor):
+    obabel_cmd_receptor = (
+        f"obabel -i pdb {receptor}.pdb -o pdbqt -O {receptor}.obabel.pdbqt"
     )
+    with subprocess.Popen(
+        obabel_cmd_receptor,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    ) as proc:
+        proc.communicate()
+
+    with subprocess.Popen(
+        f"""grep -e "^\\(ATOM\\|TER\\)" {receptor}.obabel.pdbqt \\
+            >> {receptor}.obabel.clean.pdbqt
```

Comment:
I understand that we are matching all lines with ATOM or TER for the cleaning of the pdbqt, but what does grep -e do?

Reply:
The `-e` tells `grep` to expect the pattern as the value for that parameter. I tend to think of it as less ambiguous.